### PR TITLE
osu-lazer-bin: init at 2022.1228.0

### PR DESCRIPTION
**Pull request by delan**
###### Description of changes

osu! lazer is in nixpkgs (osu-lazer), but on Linux, only the official AppImage version is allowed to submit scores or play with others (see #208019). This patch adds an osu-lazer-bin using that AppImage, based on pkgs/applications/audio/cider.

The main executable (bin/osu!) runs for me on NixOS 22.05, and the desktop file works too. I use rofi though, so I have no way of knowing if I’ve installed the icons correctly.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux (**22.05 only**)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] ~~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~~
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/osu-lazer-bin/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ appimageTools, lib, fetchurl }:
+
+appimageTools.wrapType2 rec {
+  pname = "osu-lazer-bin";
+  version = "2022.1228.0";
+
+  src = fetchurl {
+    url = "https://github.com/ppy/osu/releases/download/2022.1228.0/osu.AppImage";
```
Comment (thiagokokada) [inline pkgs/games/osu-lazer-bin/default.nix:8]:
```suggestion
    url = "https://github.com/ppy/osu/releases/download/${version}/osu.AppImage";
```

Comment (delan) [reply pkgs/games/osu-lazer-bin/default.nix]: done in 50dbb3001c3fa43e351e717b2b7c4e23c1363149

**Comment by delan** [comment]
squashed in c24775b4e915e72daf93cf985d6701b7af24e1df

File: pkgs/games/osu-lazer/default.nix [hunk 2]
```diff
@@ -77,7 +77,7 @@ buildDotnetModule rec {
   })];
 
   meta = with lib; {
-    description = "Rhythm is just a *click* away";
+    description = "Rhythm is just a *click* away (no score submission or multiplayer)";
```
Comment (thiagokokada) [inline pkgs/games/osu-lazer/default.nix:80]: Can you make this change in a separate commit?

Comment (delan) [reply pkgs/games/osu-lazer/default.nix]: removed in 411cdf44e0184556543183c697f8abb1184aa73d

**Comment by delan** [comment]
squashed in fcfdd268ca3e73022b8b5a8bf3504552b18e19bf

**Comment by thiagokokada** [comment]
@ofborg eval

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 208236` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>osu-lazer-bin</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by oxalica** [comment]
Thanks for packaging this. Score uploading works flawlessly for me.

**Comment by delan** [comment]
@huantianad i’m not sure, i guess it depends on how stable the osu client hash is across builds of the same derivation, but i don’t know how to check the client hash (the invalid client hash error doesn’t expose it).

**Comment by thiagokokada** [comment]
> This has already been merged, but was wondering, how reproducible is the original `osu-lazer` derivation? Is it possible to ask ppy to also allow the hash of the original derivation to upload scores? That way we could get score uploading even with the one from source. @delan

We tried this in the past, but upstream prefers us to ship the AppImage instead of messing with the sources: https://github.com/NixOS/nixpkgs/issues/132610

**Comment by huantianad** [comment]
Interesting conversation, I assume we weren't able to make this package reproducible and have the client hash remain consistent across builds? I see the possibility of submitting hashes mentioned, but not acted upon, as well as extracting hashes from the AppImage automatically.
